challenge.rb

```diff
@@ -1,26 +1,25 @@
-
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map {|animal| animal.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select{|animal| animal=="dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject{|animal|animal=="dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input[0..1]
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.find_all {|animal|animal[0]=="c" || animal[0]=="d"}
```

Comment:
See if [`String#start_with?`](https://ruby-doc.org/core-2.5.0/String.html#method-i-start_with-3F) simplifies this.